winlogbeat/eventlog/wineventlog.go

```diff
@@ -374,20 +401,31 @@ func newEventLogging(options *common.Config) (EventLog, error) {
 // newWinEventLog creates and returns a new EventLog for reading event logs
 // using the Windows Event Log.
 func newWinEventLog(options *common.Config) (EventLog, error) {
+	var xmlQuery string
+	var err error
+
 	c := defaultWinEventLogConfig
-	if err := readConfig(options, &c); err != nil {
+	if err = readConfig(options, &c); err != nil {
 		return nil, err
 	}
 
-	query, err := win.Query{
-		Log:         c.Name,
-		IgnoreOlder: c.SimpleQuery.IgnoreOlder,
-		Level:       c.SimpleQuery.Level,
-		EventID:     c.SimpleQuery.EventID,
-		Provider:    c.SimpleQuery.Provider,
-	}.Build()
-	if err != nil {
-		return nil, err
+	id := c.ID
+	if c.XMLQuery != "" {
+		xmlQuery = c.XMLQuery
+	} else {
+		if id == "" {
```

Comment:
`} else if id == "" {` to reduce indentation.

Reply:
We can't do that here since non-XML query configs don't require an ID (this is for backwards compatibility with existing configurations). I ended up lifting this if block outside of the else to reduce indentation.